tests/integration/hive_metastore/test_catalog_schema.py

```diff
@@ -16,6 +17,36 @@
 _SPARK_CONF = get_azure_spark_conf()
 
 
+@pytest.fixture
+def catalog_name(ws, make_random) -> Iterator[str]:
+    name = f"test-catalog-{make_random(5)}"
+    yield name
+    try:
+        ws.catalogs.delete(name, force=True)
+        logger.info(f"Deleted test catalog: {name}")
+    except NotFound:
+        pass
+
+
+@retried(on=[NotFound], timeout=timedelta(minutes=2))
+def test_create_ucx_catalog_creates_catalog(
+    ws,
+    runtime_ctx,
+    watchdog_remove_after,
+    catalog_name,
+) -> None:
+    prompts = MockPrompts({f"Please provide storage location url for catalog: {catalog_name}": "metastore"})
+    runtime_ctx.catalog_schema.UCX_CATALOG = catalog_name
+
+    runtime_ctx.catalog_schema.create_ucx_catalog(prompts, properties={"RemoveAfter": watchdog_remove_after})
+
+    @retried(on=[KeyError], timeout=timedelta(seconds=20))
+    def get_catalog(name: str) -> CatalogInfo:
+        return ws.catalogs.get(name)
```

Comment:
Is this here to deal with a consistency problem?

Reply:
Yeah, to be sure, I do not know if the `catalogs` API has a consistency problem